src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -5385,7 +5385,7 @@ std::optional<ProjectionCandidate> MergeTreeData::getQueryProcessingStageWithAgg
         if (analysis_result.before_where)
         {
             candidate.where_column_name = analysis_result.where_column_name;
-            candidate.remove_where_filter = analysis_result.remove_where_filter;
+            candidate.remove_where_filter = !required_columns.contains(analysis_result.where_column_name);
```

Comment:
Here is the fix which should be used in https://github.com/ClickHouse/ClickHouse/pull/35788